ui/app/src/Utils/FormValidation.elm

```diff
@@ -0,0 +1,62 @@
+module Utils.FormValidation
+    exposing
+        ( ValidationResult(Valid, NotValid)
+        , ValidatedString
+        , validateDate
+        , areValidResults
+            , validatedMatcherToMatcher
+        , ValidatedMatcher
+        )
+
+import Utils.Date exposing (timeFromString)
+import Silences.Types exposing (Silence)
+import Utils.Types exposing (Matcher)
+
+
+type ValidationResult
+    = Valid
+    | NotValid String
+
+
+type alias ValidatedString =
+    { value : String
+    , result : ValidationResult
+    }
```

Comment:
My only issue with `Result` is, that, in case of `Err` I still want to access a. Should I just create a tuple:
```
Result a (a, String)
```
Where `a` is my actual text input (e.g. Silence creator name) and String is my error message.

@w0rm What do you think?